packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js

```diff
@@ -30,22 +41,34 @@ export default class QuanticColoredResultBadge extends LightningElement {
   /** @type{boolean} */
   invalidColor = false;
 
+  hasValidTextContent() {
+    return !!this.label || !!this.result?.raw?.[this.field]
+  }
+
   renderedCallback() {
-    if (this.label && this.color) {
+    if (this.hasValidTextContent() && this.color) {
       this.setBadgeColors();
-    } else if (this.label) {
+    } else if (this.hasValidTextContent()) {
       this.setBadgeDefaultColors();
     }
   }
 
   /**
-   * Sets the primary and scondary colors to be used in the colored badge.
+   * get the text content to display depending the provided props.
+   * @returns {string}
+   */
+  get textToDisplay() {
+    return this.label || this.result?.raw?.[this.field].toString()
```

Comment:
Now that we have the Quantic components to display all the types of fields (string, number, date and multi value fields) I think it would be nice use these here instead of always displaying the field value as a string.

For example if the user tries to use this component with the field date, the displayed text would be a timestamp and not a human readable date.

I was going to create in SFINT the component that will take a  `field`, a `result` and a `type` as properties and will take care of displaying the field in a proper way according to the given type.
But now that I see that we can also use it here, I think it would make sense to create the component in Quantic.
 The QuanticResultField it takes like I mentioned `field`, `result` and `type` and depending on the type it renders one of the following: `QuanticResultText`, `QuanticResultNumber`, `QuanticResultDate`, `QuanticResultMultiValueText`
 What do you think @nathanlb  ?

Reply:
Component created in this PR: https://github.com/coveo/ui-kit/pull/2656 👌